showcase/showcase.ts

```diff
@@ -1,11 +1,12 @@
-import { PortableTextComponent, PortableTextImage, PortableTextInternalLink, PortableTextTable } from "../src/index.js"
+import { PortableTextComponentOrItem, PortableTextImage, PortableTextItemLink, PortableTextTable } from "../src/index.js"
   
-  const portableTextComponent: PortableTextComponent = {
-    _type: "component",
+  const portableTextComponent: PortableTextComponentOrItem = {
+    _type: "componentOrItem",
     _key: "guid",
     component: {
-      _ref: "linkedItemOrComponentCodename",
```

Comment:
@pokornyd , Would it be possible to change the `_ref` and  `_key` in various objects with the expected value? I.e. in this case you assign a  `codename`, but when using this I have no way of knowing if it's a codename/id or something else entirely. 

If these properties are necessary for some reason and can't be renamed, I think it would be worth adding new custom properties that will represent the ID/codename appropriately.

Reply:
hey @Enngage. this had already been done. the underlying `Reference` type now also contains a `referenceType` string literal, with value of `codename`, `external-id` or `id`, informing you of the nature of the value stored in `_ref`.

the showcase is updated to reflect that, but the readme embed points to main branch version of it. should get sorted after the merge :)